### PR TITLE
Cleanup lint issues & setup CI

### PR DESCRIPTION
**Pull request by WesleyKapow**
**WIP** -- Playing with CI.

### CODE REVIEW

**Review by luthfib — APPROVED** [review]
LGTM, good luck

**Review by luthfib — DISMISSED** [review]
LGTM

**Review by luthfib — DISMISSED** [review]
LGTM. Lots of good cleanup. Left a small comment

File: package.json [hunk 1]
```diff
@@ -54,6 +61,7 @@
     "@vitejs/plugin-react": "^2.0.0",
     "eslint": "^8.22.0",
     "eslint-config-prettier": "^8.5.0",
+    "happy-dom": "^6.0.4",
```
Comment (luthfib) [inline package.json:64]: Are we currently using this? Or are you anticipating we will need to mock the dom in the future for tests?

Comment (WesleyKapow) [reply package.json]: That's what we use in CTW.. I was struggling to get something working and switched to `environment: "jsdom"` in `vitest.config`.. seems to work so dropping this un-used dependency. Good catch!

**Review by luthfib — APPROVED** [review]

**Pull request merged**